src/components/Reports/GDPRPermissions/PermissionExpandableRow.js

```diff
@@ -62,36 +98,69 @@ const PermissionExpandableRow = ({ field, dependencies: { experimentalFeatures }
-                <thead>
-                  <tr>
-                    <th aria-label={_('subscriber_gdpr.consent')} scope="col">
-                      <FormattedMessage id="subscriber_gdpr.consent" tagName="span" />:
-                    </th>
-                    <th aria-label={_('subscriber_gdpr.modification_source_ip')} scope="col">
-                      <FormattedMessage
-                        id="subscriber_gdpr.modification_source_ip"
-                        tagName="span"
-                      />
-                      :
-                    </th>
-                    <th aria-label={_('subscriber_gdpr.modification_date')} scope="col">
-                      <FormattedMessage id="subscriber_gdpr.modification_date" tagName="span" />:
-                    </th>
-                    <th aria-label={_('subscriber_gdpr.source_form')} scope="col">
-                      <FormattedMessage id="subscriber_gdpr.source_form" tagName="span" />:
-                    </th>
-                  </tr>
-                </thead>
-              </table>
+          {loading ? (
+            <>
+              <td />
+              <td>
+                <S.EmptyBox>
+                  <Loading />
+                </S.EmptyBox>
+              </td>
+              <td />
+            </>
+          ) : unexpectedError ? (
+            <td className="dp-unexpected-error-table" colSpan={3}>
+              <span>
+                <span className="dp-icon-warning" /> ¡Ouch! Hubo un problema de conexión. <a href="#" onClick={reloadRowData}> Reintenta aquí</a>
+              </span>
             </td>
-            <td />
-          </>
+          ) : (
+            <>
+              <td className="dp-latest-results">
+                <FormattedMessage id="subscriber_gdpr.latest_results" tagName="span" />
+              </td>
+              <td className="dp-list-results">
+                <table className="dp-table-results">
+                  <thead>
+                    <tr>
+                      <th aria-label={_('subscriber_gdpr.consent')} scope="col">
+                        <FormattedMessage id="subscriber_gdpr.consent" tagName="span" />:
+                      </th>
+                      <th aria-label={_('subscriber_gdpr.modification_source_ip')} scope="col">
+                        <FormattedMessage
+                          id="subscriber_gdpr.modification_source_ip"
+                          tagName="span"
+                        />
+                        :
+                      </th>
+                      <th aria-label={_('subscriber_gdpr.modification_date')} scope="col">
+                        <FormattedMessage id="subscriber_gdpr.modification_date" tagName="span" />:
+                      </th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    {permissions.length &&
+                      permissions.map(({ value, originIP, date }, index) => {
```

Comment:
Is checking length necessary?

If I am not wrong it will be the same:

```suggestion
                    {permissions.map(({ value, originIP, date }, index) => {
```

Reply:
Am I missing something?

Reply:
You are right, it is not necessary.